### DISCUSSION TITLE
Error with

### DISCUSSION THREAD

**Discussion by surf123** [wheels-dev/wheels#1465] [Q&A]
Attempting to move an CF Wheels app from CF2014 to CF2021 and have a bug that may be out of my wheelhouse. App worked on a hosted CF2014 server without issues. App is being moved to a CF2021 environment and when tested it failed while processing the appfunctions.cfm file on the CFInclude call on the file wheels/global/cfml.cfm. 

The error message is:
Cannot find implementation class
coldfusion.tagext.image.ImageTag for the image Tag.

It fails on the <cfimage> line which makes sense.

```coldfusion
<cffunction name="$image" returntype="struct" access="public" output="false">
	<cfset var rv = {}>
	<cfset arguments.structName = "rv">
	<cfimage attributeCollection="#arguments#">
	<cfreturn rv>
</cffunction> 
``` 

If I comment this function out it fails in the next < cffunction> which is named $Mail.  

I built a local VM and did a clean install of Windows Server and CF2021. The app works in that VM without any issues. Same version of Windows Server. Any help will be much appreciated.

**Comment by neokoenig** [comment]
Might be related to Adobe's Package Manager which was introduced in CF2021?
https://helpx.adobe.com/uk/coldfusion/using/coldfusion-package-manager.html

Sounds like locally you might have everything installed, but missing some packages in prod.

**Comment by surf123** [comment]
@neokoenig you are definitely on to something. As it turns out there are 5 other applications moving from a working CF2021 install to a CF2021 install. None of the others use CFWheels, but 3 of them are getting similar implantation class errors. All are working without issue on the production server (original install of CF2016 that was upgraded in place to CF2021). 

The test server is a clean VM that has CF2021 and never had CF2016. Interestingly enough my desktop VM was CF2016 and upgraded to CF2021 and works. Both the production server and the test server have identical packages so I am doubting that uninstalled packages are the issue. 

My local VM has all of the packages so I am going to start deleting them to see if there could be one that is causing it.

I also am going to start with a clean VM and do an install of CF2021 on it to eliminate the possibility of some baggage being left behind after the upgrade.
